src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -5717,6 +5717,92 @@ static void Main()
             Assert.Equal(SymbolKind.Discard, isymbol.Kind);
         }
 
+        [Fact, WorkItem(25829, "https://github.com/dotnet/roslyn/issues/25829")]
+        public void SameTypeDiscardsAreEqual()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        (_, _) = (1, 2);
+        _ = 3;
+        M(out _);
+    }
+    static void M(out int x) => x = 1;
+}
+";
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var discards = GetDiscardIdentifiers(tree).ToArray();
+            Assert.Equal(4, discards.Length);
+            var symbol0 = (IDiscardSymbol)model.GetSymbolInfo(discards[0]).Symbol;
+            var set = new HashSet<ISymbol>();
+            foreach (var discard in discards)
+            {
+                var symbol = (IDiscardSymbol)model.GetSymbolInfo(discard).Symbol;
+                set.Add(symbol);
```

Comment:
>set.Add(symbol); [](start = 16, length = 16)

It would be good to assert that all the symbols are distinct reference-wise.

Reply:
Including the ```symbol0```.

---
In reply to: [186805636](https://github.com/dotnet/roslyn/pull/26720#discussion_r186805636) [](ancestors = 186805636)

Reply:
Such an assertion would not be correct.

Reply:
These symbols are not specified to be distinct reference-wise, and adding a test constraining the implementation in that way would be counterproductive.

Reply:
I argue that this is important to confirm the value of this test. Verifying equality of an instance to itself is not the goal of this test. Quite the opposite and we should clearly assert this fact.

---
In reply to: [186806231](https://github.com/dotnet/roslyn/pull/26720#discussion_r186806231) [](ancestors = 186806231,186805636)

Reply:
> These symbols are not specified to be distinct reference-wise, and adding a test constraining the implementation in that way would be counterproductive.

Quite the opposite. It will make it easy to confirm that the test is testing (and will continue doing so in the future) scenario it is supposed to test.

---
In reply to: [186881434](https://github.com/dotnet/roslyn/pull/26720#discussion_r186881434) [](ancestors = 186881434)

Reply:
The scenario the test is supposed to test is not the identity of the symbols, but their equality. And it does that, in both positive and negative scenarios. It is not intended to enforce that the discards are reference unequal, and in fact I can easily imagine that that would change in the future. The intended behavior is the behavior that is tested.

---
In reply to: [186883163](https://github.com/dotnet/roslyn/pull/26720#discussion_r186883163) [](ancestors = 186883163,186881434)